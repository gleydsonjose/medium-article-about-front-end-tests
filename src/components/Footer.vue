<template>
  <footer class="footer">
    <section class="footer__social-media-group">
      <a
        class="footer__social-media footer__social-media--facebook"
        target="_blank"
        href="https://pt-br.facebook.com"
      >
        <i class="mdi mdi-facebook"></i>
      </a>
      <a
        class="footer__social-media footer__social-media--whatsapp"
        target="_blank"
        href="https://web.whatsapp.com"
      >
        <i class="mdi mdi-whatsapp"></i>
      </a>
      <a
        class="footer__social-media footer__social-media--twitter"
        target="_blank"
        href="https://twitter.com/"
      >
        <i class="mdi mdi-twitter"></i>
      </a>
      <a
        class="footer__social-media footer__social-media--instagram"
        target="_blank"
        href="https://www.instagram.com/"
      >
        <i class="mdi mdi-instagram"></i>
      </a>
    </section>

    <p class="footer__copyright-text">
      Copyright Â© 2022 Tests. All rights reserved.
    </p>
  </footer>
</template>

<script>
export default {
  name: "Footer",
};
</script>

<style scoped>
.footer {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  background-color: #222222;
  height: 120px;
  width: 100%;
}

.footer__social-media-group {
  padding-bottom: 15px;
}

.footer__social-media {
  font-size: 25pt;
  text-decoration: none;
  margin-left: 10px;
  transition-duration: 0.3s;
  transition-timing-function: ease-out;
  transition-property: filter;
}

.footer__social-media:hover {
  filter: grayscale(1);
  transition-duration: 0.3s;
  transition-timing-function: ease-in;
  transition-property: filter;
}

.footer__social-media--facebook {
  color: #1877f2;
}

.footer__social-media--whatsapp {
  color: #26cc64;
}

.footer__social-media--twitter {
  color: #1c96e8;
}

.footer__social-media--instagram {
  background-image: linear-gradient(to bottom, #c92e82, #ef7a2a);
  background-clip: text;
  -webkit-text-fill-color: transparent;
}

.footer__copyright-text {
  font-size: 10pt;
  margin: 0;
  color: #d4d4d4;
}
</style>
