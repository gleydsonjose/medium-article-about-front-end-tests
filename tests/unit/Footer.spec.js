import { shallowMount } from "@vue/test-utils";
import Footer from "@/components/Footer.vue";

describe("<template/>", () => {
  it("deveria ser igual ao snapshot", () => {
    const wrapper = shallowMount(Footer);

    expect(wrapper).toMatchSnapshot();
  });
});
