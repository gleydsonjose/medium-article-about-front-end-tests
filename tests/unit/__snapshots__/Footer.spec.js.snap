// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<template/> deveria ser igual ao snapshot 1`] = `
<footer class="footer">
  <section class="footer__social-media-group"><a target="_blank" href="https://pt-br.facebook.com" class="footer__social-media footer__social-media--facebook"><i class="mdi mdi-facebook"></i></a> <a target="_blank" href="https://web.whatsapp.com" class="footer__social-media footer__social-media--whatsapp"><i class="mdi mdi-whatsapp"></i></a> <a target="_blank" href="https://twitter.com/" class="footer__social-media footer__social-media--twitter"><i class="mdi mdi-twitter"></i></a> <a target="_blank" href="https://www.instagram.com/" class="footer__social-media footer__social-media--instagram"><i class="mdi mdi-instagram"></i></a></section>
  <p class="footer__copyright-text">
    Copyright Â© 2022 Tests. All rights reserved.
  </p>
</footer>
`;
